io.go

```diff
@@ -89,6 +90,32 @@ func NewIOWithTerminal(stdin io.Reader, stdout, stderr io.Writer, terminal bool)
 	}
 }
 
+//NewIOWithFifoSet returns IOCreation handler function for a custom fifo path
+func NewIOWithFifoSet(stdin io.Reader, stdout, stderr io.Writer, paths *FIFOSet, terminal bool) IOCreation {
```

Comment:
In the context of swarmkit's containerd executor I had the same issue. In that code I currently open `/dev/null` manually and pass it as the first argument to `NewIOWithTerminal` but I also considered that it might be nicer if `NewIOWithTerminal` would turn a `nil` argument for any of the stdio arguments into `/dev/null` for you.

I'm assuming here that when you say "pass empty stdin fifo path" the desired result is the containers has `/dev/null` on fd 0, rather than having fd 0 not be open at all.